stmtcacher.go

```diff
@@ -66,6 +80,32 @@ func (sc *stmtCacher) QueryRow(query string, args ...interface{}) RowScanner {
 	return stmt.QueryRow(args...)
 }
 
+// Close closes all prepared statements in the statement cache
+// and should not be used by subsequent callers
+// it does not close the underlying db being proxied
+func (sc *stmtCacher) Close() error {
```

Comment:
Given the nature of this type as a `DB` proxy, I wouldn't implement `Close` to mean something different from `DB.Close`. Maybe `Clear`?

Reply:
In addition, you wouldn't need the `close` field or `ErrStmtCacheClosed`; you could just close all the cached stmts and clear the cache map.

Reply:
I like it 💯